### PR TITLE
Docs/waf vault tf

### PR DESCRIPTION
**Pull request by cjobermaier**
Migrate [programmatic resource management best practices](https://developer.hashicorp.com/well-architected-framework/operational-excellence/operational-excellence-managing-vault-with-terraform) from WAF to Vault product docs.

Usage guide for migrating unmanaged resources to follow in another PR.

### CODE REVIEW

File: website/content/docs/configuration/programmatic-best-practices.mdx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+</Note>
+
+## Vault recommendations
+
+- **Use Vault-backed dynamic credentials for different cloud providers**.
+  Vault-backed dynamic credentials limits the need for unique dynamic provider
+  credentials across different cloud providers so you can centralize sensitive
+  data management with Vault while generating short-lived credentials with
+  multiple cloud providers.
```
Comment (cstella84) [inline website/content/docs/configuration/programmatic-best-practices.mdx:75]: Can we link to this page?

https://developer.hashicorp.com/terraform/cloud-docs/workspaces/dynamic-provider-credentials/vault-backed

File: website/content/docs/configuration/programmatic-best-practices.mdx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+- **Use distinct token capabilities**. Use the minimal required Vault token
+  capabilities to manage your Vault resources. For example, pulling data from a
+  KV plugin only requires the `read` capability while drift detection requires
+  `read`, `create`, and `update`.
+
+- **Migrate existing resources to Terraform**. If you created Vault resources
+  outside of your Terraform provisioning workflow,
+  [migrate the un-managed resources](/vault/docs/configuration/programmatic-management#migrate)
+  as soon as possible.
```
Comment (cstella84) [inline website/content/docs/configuration/programmatic-best-practices.mdx:89]: The "as soon as possible" feels unnecessary here.

File: website/content/docs/configuration/programmatic-best-practices.mdx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  multiple cloud providers.
+
+- **Use the namespace attribute in resources and data sources**. Using the
+  namespace attribute of a resource, instead of a provider alias, simplifies
+  your configuration and avoids the need for multiple provider blocks.
+
+- **Use distinct token capabilities**. Use the minimal required Vault token
+  capabilities to manage your Vault resources. For example, pulling data from a
+  KV plugin only requires the `read` capability while drift detection requires
+  `read`, `create`, and `update`.
```
Comment (cstella84) [inline website/content/docs/configuration/programmatic-best-practices.mdx:84]: See my suggestion below. The verbiage can be polished, but this is the corrected information.

```suggestion
- **Use distinct token capabilities**. Use the minimal required Vault token
  capabilities to manage your Vault resources. For example, reading data from a
  KV datasource and drift detection functionality only require the `read` capability. The `create` or `update` capability is required to manage the KV resource and depends on whether the resource already exists. The `delete` capability is required if you remove the KV resource from the Terraform configuration.
```

**Review by cstella84 — CHANGES_REQUESTED** [review]
I've added some updates, but overall, I don't see the need to separate "Terraform Vault provider recommendations" from "Vault recommendations". All of the recommendations, including those under "Vault recommendations" are related to using Vault with Terraform.

File: website/content/docs/configuration/programmatic-best-practices.mdx (hunk 1 above)
Comment (schavis) [reply website/content/docs/configuration/programmatic-best-practices.mdx:75]: Added

File: website/content/docs/configuration/programmatic-best-practices.mdx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+- **Use short-lived credentials** Credentials persist in the Terraform state
+  file. Short-lived credentials reduce the risk of exposure in the event of a
+  compromised state file.
+
+## Terraform Vault provider recommendations
+
+- **Use dynamic provider credentials for Vault when possible**. Dynamic provider
+  credentials generate short-lived credentials as needed, which limits the need
+  for static credentials and improves the security posture of your integration.
```
[the remark this answers is not in the document]
Comment (schavis) [reply website/content/docs/configuration/programmatic-best-practices.mdx:49]: Added

**Review by schavis — APPROVED** [review]

**Review by cstella84 — APPROVED** [review]
LGTM

**Pull request merged**